src/UpdatePassword.js

```diff
@@ -0,0 +1,148 @@
+        const inputs = ['oldPassword', 'newPassword', 'confirmPassword'];
+        for(var i =0; i <inputs.length; i++) {
+          if(!this.state[inputs[i]])
+            return false;
+        }
+        return true;
+    }
+
+    checkPasswords() {
+        const newPassword = this.state.newPassword;
+        const confirmPassword = this.state.confirmPassword;
+        const oldPassword = this.state.oldPassword;
+        if (newPassword.length === 0) {
+          alert("Please add password");
+          return false;
+        } else if (confirmPassword != newPassword) {
+          alert("Please make sure the new passwords match");
+            
+          return false;
+        } else if ((confirmPassword == newPassword) && (newPassword == oldPassword)){
+            alert("Your new password is the same as your old password")
+            return false;
+        }
+        return true;
+    }
+        
+    handleSubmit = (event) => {
+        event.preventDefault();
+        
+        if (this.completedInput()){
+            if (this.checkPasswords()){
+                const id = this.state.userId;
+                const myForm = new FormData (document.getElementById("newPasswordForm"));
+                myForm.append("id", id);
+
+                const myRequest = new Request("http://127.0.0.1:5000/updatePassword", {
+                    method: "POST",
+                    body: myForm,
+                });
+  
+                fetch(myRequest)
+                .then((res) =>
+                res.json())
+                .then((res) => {
+                    this.setState({error: ""});
+                    if(res.response === "Success"){
+                        this.setState({
+                            error: "Successfully updated password!",
+                            });
+                    }
+                    
+                    else {
+                        this.setState({
+                            error: res.response,
+                        });
+                    }  
+                })
+                .catch((error) => {
+                    this.setState({
+                    error: "Error connecting to backend",
+                    });
+                });
+            }
+        } 
+        else{ 
+          alert("Please enter all fields");
+        }
+    };
+
+    render() {
+        return (
+            <div className="updatePassword">
+            <Header id={this.state.userId}/>
+            <form id="newPasswordForm" onSubmit={this.handleSubmit}>
+            <h1>Change Password</h1>
+                <p>Please enter your current password</p> 
+              <input 
+                name="oldPassword" 
+                type="password" 
+                value={this.state.oldPassword}
+                placeholder="current password"
+                onChange={this.handleInputChange} 
+              />
+              <br></br>
+              <p>Please enter your new password</p>
```

Comment:
should we have a text somewhere that encourages users to enter a complex enough password (ie. At least one capital letter, one punctuation, etc)?

Reply:
we could, but so far we don't have that either in register page, so for consistency, I don't think I should add one here for this sprint, for next sprint we can add that not only as a text, but maybe as a method to check if it is complicated enough before letting them register and change password.